app/src/main/java/com/sakaimobile/development/sakaiclient20/ui/adapters/AnnouncementsAdapter.java

```diff
@@ -126,100 +106,58 @@ public void onClick(View v) {
+        int subjCode = siteIdToCourse.get(currAnnouncement.siteId).subjectCode;
+        holder.courseIcon.setText(RutgersSubjectCodes.mapCourseCodeToIcon.get(subjCode));
 
-            int subjCode = siteIdToCourse.get(currAnnouncement.siteId).subjectCode;
-            announcementHolder.courseIcon.setText(RutgersSubjectCodes.mapCourseCodeToIcon.get(subjCode));
+        holder.date.setText(currAnnouncement.getShortFormattedDate());
 
-            announcementHolder.date.setText(currAnnouncement.getShortFormattedDate());
+        //check to see the announcement type
+        if (announcementType == AnnouncementsFragment.ALL_ANNOUNCEMENTS) {
 
-            //check to see the announcement type
-            if(announcementType == AnnouncementsFragment.ALL_ANNOUNCEMENTS) {
+            // if all announcements, show course title, then announcement title
+            holder.cardHeading2.setText(siteIdToCourse.get(currAnnouncement.siteId).title);
+            holder.cardHeading3.setText(currAnnouncement.title);
 
-                // if all announcements, show course title, then announcement title
-                announcementHolder.cardHeading2.setText(siteIdToCourse.get(currAnnouncement.siteId).title);
-                announcementHolder.cardHeading3.setText(currAnnouncement.title);
+        } else if (announcementType == AnnouncementsFragment.SITE_ANNOUNCEMENTS) {
 
-            }
-            else if(announcementType == AnnouncementsFragment.SITE_ANNOUNCEMENTS) {
-
-                //if site announcements, show announcement title, then announcement body
-                announcementHolder.cardHeading2.setText(currAnnouncement.title);
-
-                try {
-                    //if its after android N, use this method for setting the html
-                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
-                        announcementHolder.cardHeading3.setText(Html.fromHtml(currAnnouncement.body, Html.FROM_HTML_MODE_COMPACT));
-                    } else {
-                        announcementHolder.cardHeading3.setText(Html.fromHtml(currAnnouncement.body));
-                    }
-                } catch(RuntimeException e) {
-//                    java.lang.RuntimeException: PARAGRAPH span must start at paragraph boundary (832 follows  )
-                    announcementHolder.cardHeading3.setText("");
-                    e.printStackTrace();
-                }
+            //if site announcements, show announcement title, then announcement body
+            holder.cardHeading2.setText(currAnnouncement.title);
 
+            try {
+                //if its after android N, use this method for setting the html
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+                    holder.cardHeading3.setText(Html.fromHtml(currAnnouncement.body, Html.FROM_HTML_MODE_COMPACT));
+                } else {
+                    holder.cardHeading3.setText(Html.fromHtml(currAnnouncement.body));
```

Comment:
What are `cardHeading1`, `cardHeading2`, and `cardHeading3`? This is a bit hard to follow, perhaps rename them to be more representative of the information shown in the views.